src/commands/ssh/startSsh.ts

```diff
@@ -0,0 +1,99 @@
+    if (!node) {
+        node = <SiteTreeItem>await ext.tree.showTreeItemPicker(WebAppTreeItem.contextValue);
+    }
+
+    const currentSshTerminal: sshTerminal | undefined = sshSessionsMap.get(node.root.client.fullName);
+    if (currentSshTerminal && currentSshTerminal.running) {
+        throw new Error(`Azure SSH is currently starting or already started for "${node.root.client.fullName}".`);
+    }
+
+    try {
+        sshSessionsMap.set(node.root.client.fullName, { running: true, terminal: undefined });
+        await startSshInternal(node);
+    } catch (error) {
+        sshSessionsMap.set(node.root.client.fullName, { running: false, terminal: undefined });
+        throw error;
+    }
+}
+
+async function startSshInternal(node: SiteTreeItem): Promise<void> {
+    const siteClient: SiteClient = node.root.client;
+    const siteConfig: SiteConfigResource = await siteClient.getSiteConfig();
+
+    // should always be an unbound port
+    const localHostPortNumber: number = await portfinder.getPortPromise();
+    const sshPortNumber: number = 2222;
+    const confirmDisableMessage: string = 'The app configuration will be updated to disable remote debugging and restarted. Would you like to continue?';
+
+    await vscode.window.withProgress({ location: vscode.ProgressLocation.Notification }, async (progress: vscode.Progress<{}>): Promise<void> => {
+        if (!siteClient.isLinux) {
+            throw new Error('Azure SSH is only supported for Linux web apps.');
+        }
+
+        remoteDebug.reportMessage('Checking app settings...', progress);
+
+        // remote debugging has to be disabled in order to tunnel to the 2222 port
+        await remoteDebug.setRemoteDebug(false, confirmDisableMessage, undefined, siteClient, siteConfig);
+
+        remoteDebug.reportMessage('Initializing SSH...', progress);
+        const publishCredential: User = await siteClient.getWebAppPublishCredential();
+        const tunnelProxy: TunnelProxy = new TunnelProxy(localHostPortNumber, sshPortNumber, siteClient, publishCredential);
+
+        await tunnelProxy.startProxy();
+
+        remoteDebug.reportMessage('Connecting to SSH...', progress);
+        await connectToTunnelProxy(tunnelProxy);
+    });
+
+    async function connectToTunnelProxy(tunnelProxy: TunnelProxy): Promise<void> {
+        const sshTerminalName: string = `${node.root.client.fullName} - SSH`;
+        // -o StrictHostKeyChecking=no doesn't prompt for adding to hosts
+        // -o "UserKnownHostsFile /dev/null" doesn't add host to known_user file
+        // -o "LogLevel ERROR" doesn't display Warning: Permanently added 'hostname,ip' (RSA) to the list of known hosts.
+        const sshCommand: string = `ssh -c aes256-cbc -o StrictHostKeyChecking=no -o "UserKnownHostsFile /dev/null" -o "LogLevel ERROR" root@127.0.0.1 -p ${localHostPortNumber}`;
+        const terminal: vscode.Terminal = vscode.window.createTerminal(sshTerminalName);
```

Comment:
Have you tried the following?:
1. Start SSH
1. Stop SSH
1. Start SSH

I'm just curious if `vscode.window.createTerminal` works if the terminal with that name already exists

Reply:
Yeah, I have.  Stop SSH will dispose of the terminal.  If you try to Stop SSH before it's actually connected, it's ignored rather than canceling the SSHing.